src/lib/features/dependent-features/dependent-features-service.ts

```diff
@@ -90,6 +90,10 @@ export class DependentFeaturesService {
     ): Promise<void> {
         const { enabled, feature: parent, variants } = dependentFeature;
 
+        if (child === parent) {
+            throw new InvalidOperationError('Self dependency not allowed.');
```

Comment:
Wonder if we can make this slightly clearer, like this? But it's very minor.

```suggestion
            throw new InvalidOperationError('A feature flag cannot depend on itself.');
```